cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/FindPluginHelper.java

```diff
@@ -74,8 +73,7 @@ public static Plugin findPlugin(ArtifactRepository artifactRepository,
     // Just verify if all required properties are provided.
     // No type checking is done for now.
     for (PluginPropertyField field : pluginEntry.getValue().getProperties().values()) {
-      Preconditions.checkArgument(!field.isRequired() ||
-                                    (properties != null && properties.getProperties().containsKey(field.getName())),
+      Preconditions.checkArgument(!field.isRequired() || (properties.getProperties().containsKey(field.getName())),
```

Comment:
is it safe to remove that null check?  Wondering whether it was just mistakenly not marked as nullable.